res/controllers/Roland_DJ-202-scripts.js

```diff
@@ -227,7 +227,60 @@ DJ202.Deck = function (deckNumbers, offset) {
     // ============================= TRANSPORT ==================================
     this.play = new components.PlayButton([0x90 + offset, 0x00]); // LED doesn't stay on
     this.cue = new components.CueButton([0x90 + offset, 0x01]);
-    this.sync = new components.SyncButton([0x90 + offset, 0x02]); // doesn't work properly
+    var SyncButton = function (options) {
+        components.SyncButton.call(this, options);
+    };
+
+    SyncButton.prototype = new components.SyncButton({
+        doubleTapTimeout: 500,
+        unshift: function () {
+            this.input = function (channel, control, value, status, group) {
+                if (this.isPress(channel, control, value, status)) {
+                    if (this.isDoubleTap) {
+                        var fileBPM = engine.getValue(this.group, 'file_bpm');
+                        engine.setValue(this.group, 'bpm', fileBPM);
+                    } else if (engine.getValue(this.group, 'sync_enabled') === 0) {
+                        engine.setValue(this.group, 'beatsync', 1);
+                        this.longPressTimer = engine.beginTimer(this.longPressTimeout, function () {
+                            engine.setValue(this.group, 'sync_enabled', 1);
+                            this.longPressTimer = 0;
+                        }, true);
+                        this.isDoubleTap = true; // For the next call.
+                        this.doubleTapTimer = engine.beginTimer(this.doubleTapTimeout, function () {
+                            this.isDoubleTap = false;
+                        }, true);
+                    } else {
+                        engine.setValue(this.group, 'sync_enabled', 0);
```

Comment:
I just experienced a glitch after a double tap while sync was enabled. Can't reproduce it right now, so just leaving a note to maybe investigate this later.